path_roles.go

```diff
@@ -352,8 +466,29 @@ func (b *azureSecretBackend) pathRoleList(ctx context.Context, req *logical.Requ
 
 func (b *azureSecretBackend) pathRoleDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	name := d.Get("name").(string)
+	role, err := getRole(ctx, name, req.Storage)
+	if err != nil {
+		return nil, fmt.Errorf("error getting role: %w", err)
+	}
+
+	if role != nil && role.PersistApp {
+		c, err := b.getClient(ctx, req.Storage)
+		if err != nil {
+			return nil, fmt.Errorf("error during delete: %w", err)
+		}
+
+		// unassigning roles is effectively a garbage collection operation.
+		c.unassignRoles(ctx, role.RaIDs)
```

Comment:
I could do what is done https://github.com/hashicorp/vault-plugin-secrets-azure/blob/2567884f30722190dbdcf5031352c58046a14bee/path_service_principal.go#L252 and return a warning in the response. I don't know what the behavior of returning a response from this function since it currently returns nil.